dataactvalidator/filestreaming/loaderUtils.py

```diff
@@ -25,85 +28,62 @@ def compareRecords (recordA,recordB, fields) :
-                    options = fieldOptions[field]
-                    if "strip_commas" in options:
-                        # Remove commas from numeric fields
-                        row[field] = row[field].replace(",","")
-                        if row[field] == "":
-                            # If empty, set to 0
-                            row[field] = "0"
-                    if "pad_to_length" in options:
-                        padLength = options["pad_to_length"]
-                        row[field] = Validator.padToLength(row[field],padLength)
-                    if "skip_duplicates" in options:
-                        if row[field] is None or len(row[field].strip()) == 0 or row[field] in valuePresent[field]:
-                            # Value not provided or already exists, skip it
-                            skipInsert = True
-                        else:
-                            # Insert new value
-                            valuePresent[field][row[field]] = True
-                record = model(**row)
-                if not skipInsert:
-                    try:
-                        interface.session.merge(record)
-                        interface.session.commit()
+        # toss out blank rows
+        data.dropna(inplace=True, how='all')
+
+        # Fix up cells that have spaces instead of being empty.
+        # Set the truly empty cells to None so they get inserted to db as NULL
+        # TODO: very ugly function below...is there a better way?
+        data.applymap(lambda x: str(x).strip() if len(str(x).strip()) else None)
+
+        # clean the dataframe column names
+        data.rename(columns=lambda x: cls.cleanColNamesFunction(x), inplace=True)
+        # make sure all values in fieldMap parameter are in the dataframe/csv file
+        for field in fieldMap:
+            if field not in list(data.columns):
+                raise ValueError("{} is required for loading table{}".format(field, model))
+        # toss out any columns from the csv that aren't in the fieldMap parameter
+        data = data[list(fieldMap.keys())]
+        # rename columns as specified in fieldMap
+        data = data.rename(columns=fieldMap)
 
-                    except IntegrityError as e:
-                        # Hit a duplicate value that violates index, skip this one
-                        print("".join(["Warning: Skipping this row: ",str(row)]))
-                        print("".join(["Due to error: ",str(e)]))
-                        interface.session.rollback()
-                        continue
-            interface.session.commit()
+        # apply column options as specified in fieldOptions param
+        for col, options in fieldOptions.items():
+            if "pad_to_length" in options:
+                # pad to specified length
+                data['{}'.format(col)] = data['{}'.format(col)].apply(
+                    lambda x: cls.padFunction(x, options['pad_to_length']))
```

Comment:
NBD, but if we're passing around lots of functions and lambdas, it might make sense to curry the parameters. For example:
```python
def mk_pad_fn(length):
    return lambda field: str(field).strip().zfill(length)
# ...
data[...] = data[...].apply(mk_pad_fn(options['pad_to_length']))
```

This should also possible with `functools.partial`, though it feels less readable:
```python
data[...] = data[...].apply(partial(
    cls.padFunction, padTo=options['pad_to_length']))
```